File: codegen/results/v002/gemma4_31b_think/20260414_075045/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * The program reads the file `input/text.txt`, cleans the text by keeping only letters,
 * counts the frequency of each lowercase word, and outputs them sorted by frequency
 * (descending) and then alphabetically (ascending).
 */
function solve() {
    const filePath = path.join('input', 'text.txt');
    
    let text: string;
    try {
        text = fs.readFileSync(filePath, 'utf8');
    } catch (err) {
        // If file doesn't exist or cannot be read, exit silently or handle as needed.
        // Given the prompt, we assume the file exists.
        return;
    }

    // 1. Convert to lowercase and find all sequences of letters (stripping punctuation/numbers)
    const words = text.toLowerCase().match(/[a-z]+/g) || [];

    // 2. Count frequencies using a Map
    const frequencyMap = new Map<string, number>();
    for (const word of words) {
        frequencyMap.set(word, (frequencyMap.get(word) || 0) + 1);
    }

    // 3. Convert Map to an array for sorting
    const sortedWords = Array.from(frequencyMap.entries());

    // 4. Sort by count descending, then alphabetically ascending
    sortedWords.sort((a, b) => {
        const [wordA, countA] = a;
        const [wordB, countB] = b;

        if (countB !== countA) {
            return countB - countA; // Descending frequency
        }
        return wordA.localeCompare(wordB); // Ascending alphabetical
    });

    // 5. Output result in "word: count" format
    for (const [word, count] of sortedWords) {
        process.stdout.write(`${word}: ${count}\n`);
    }
}

solve();

File: codegen/results/v002/gemma4_31b_think/20260414_075045/python/002_word_frequency/solution.py
import re
from collections import Counter

def main():
    try:
        # Read the input file
        with open('input/text.txt', 'r', encoding='utf-8') as file:
            text = file.read()

        # Convert to lowercase
        text = text.lower()

        # Use regex to find all sequences of letters (stripping all punctuation/digits)
        words = re.findall(r'[a-z]+', text)

        # Count the frequency of each word
        counts = Counter(words)

        # Sort the results:
        # 1. By count descending (-x[1])
        # 2. By word ascending (x[0])
        sorted_words = sorted(counts.items(), key=lambda x: (-x[1], x[0]))

        # Output each word and its count
        for word, count in sorted_words:
            print(f"{word}: {count}")
            
    except FileNotFoundError:
        pass

if __name__ == "__main__":
    main()